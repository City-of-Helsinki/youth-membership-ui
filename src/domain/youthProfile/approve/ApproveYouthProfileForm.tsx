import React from 'react';
import { useTranslation } from 'react-i18next';
import { Field, Form, Formik } from 'formik';
import { differenceInYears } from 'date-fns';
import { Button, RadioButton } from 'hds-react';

import {
  CreateAdditionalContactPersonInput,
  UpdateAdditionalContactPersonInput,
} from '../../../graphql/generatedTypes';
import convertDateToLocale from '../../../common/helpers/convertDateToLocale';
import LabeledValue from '../../../common/components/labeledValue/LabeledValue';
import Text from '../../../common/components/text/Text';
import Stack from '../../../common/components/stack/Stack';
import TermsField from '../../../common/components/termsField/TermsField';
import BasicInformationGrid from '../../../common/components/basicInformationGrid/BasicInformationGrid';
import InfoGrid from '../../../common/components/infoGrid/InfoGrid';
import YouthProfileApproverFields from '../../youthProfile/form/YouthProfileApproverFields';
import { approveYouthProfileFormSchema } from '../youthProfileSchemas';
import ageConstants from '../constants/ageConstants';
import styles from './approveYouthProfileForm.module.css';

export type FormValues = {
  firstName: string;
  lastName: string;
  address: string;
  addresses: string[];
  email: string;
  phone: string;
  language: string;
  birthDate: string;
  schoolName: string;
  schoolClass: string;
  approverFirstName: string;
  approverLastName: string;
  approverPhone: string;
  approverEmail: string;
  photoUsageApproved: string;
  languageAtHome: string;
  additionalContactPersons: (
    | CreateAdditionalContactPersonInput
    | UpdateAdditionalContactPersonInput
  )[];
};

type Props = {
  profile: FormValues;
  onSubmit: (values: FormValues) => Promise<unknown>;
};

function ApproveYouthProfileForm(props: Props) {
  const { t } = useTranslation();

  const age = differenceInYears(new Date(), new Date(props.profile.birthDate));
  const birthDate = convertDateToLocale(props.profile.birthDate);
  const schoolInfo = [props.profile.schoolName, props.profile.schoolClass]
    .filter(value => value)
    .join(', ');

  return (
    <Formik
      initialValues={{
        ...props.profile,
        terms: false,
      }}
      onSubmit={({ terms, ...values }) => props.onSubmit(values)}
      validationSchema={approveYouthProfileFormSchema}
      enableReinitialize={true}
    >
      {props => (
        <Stack space="xl">
          <div>
            <Text variant="h1">{t('approval.title')}</Text>
            <Text variant="info">
              {props.values.firstName} {t('approval.formExplanationText_1')}{' '}
              {props.values.firstName} {t('approval.formExplanationText_2')}
            </Text>
          </div>
          <div>
            <Text variant="h2">{t('approval.basicInfo')}</Text>
            <BasicInformationGrid
              name={`${props.values.firstName} ${props.values.lastName}`}
              addresses={[props.values.address, ...props.values.addresses]}
              email={props.values.email}
              phone={props.values.phone}
              birthDate={birthDate}
              language={props.values.language}
            />
          </div>
          <div>
            <Text variant="h2">{t('approval.addInfo')}</Text>
            <InfoGrid>
              <LabeledValue
                label={t('approval.schoolInfo')}
                value={schoolInfo}
                noMargin
              />
            </InfoGrid>
          </div>
          <Form>
            <Stack space="xl">
              <div>
                {age < ageConstants.PHOTO_PERMISSION_MIN && (
                  <React.Fragment>
                    <Text variant="h2">{t('approval.approverAcceptance')}</Text>
                    <Text variant="h3">{t('approval.photoUsageApproved')}</Text>
                    <Text variant="info">
                      {t('approval.photoUsageApprovedText')}
                    </Text>
                  </React.Fragment>
                )}
                {age < ageConstants.PHOTO_PERMISSION_MIN && (
                  <div className={styles.formFields}>
                    <ul className={styles.list}>
                      <li className={styles.radioButtonRow}>
                        <Field
                          id="photoUsageApprovedYes"
                          name="photoUsageApproved"
                          as={RadioButton}
                          labelText={t('approval.photoUsageApprovedYes')}
                          value="true"
                          checked={props.values.photoUsageApproved === 'true'}
                        />
                      </li>
                      <li className={styles.radioButtonRow}>
                        <Field
                          id="photoUsageApprovedNo"
                          name="photoUsageApproved"
                          as={RadioButton}
                          labelText={t('approval.photoUsageApprovedNo')}
                          value="false"
                          checked={props.values.photoUsageApproved === 'false'}
                        />
                      </li>
                    </ul>
                  </div>
                )}
              </div>
              <div>
                <Text variant="h2">{t('approval.approverInfo')}</Text>
                <Text variant="info">{t('approval.approverInfoText')}</Text>
                <YouthProfileApproverFields
                  additionalContactPersonHelperText={t(
                    'approval.addGuardianText'
                  )}
                  viewer="approver"
                />
              </div>
              <TermsField id="terms" name="terms" />
              <Button className={styles.button} type="submit">
                {t('approval.approveButton')}
              </Button>
            </Stack>
          </Form>
        </Stack>
      )}
    </Formik>
  );
}

export default ApproveYouthProfileForm;
