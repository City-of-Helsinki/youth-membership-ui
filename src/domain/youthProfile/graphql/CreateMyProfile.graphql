mutation CreateMyProfile($input: CreateMyProfileMutationInput!) {
  createMyProfile(input: $input) {
    profile {
      id
    }
  }
}
