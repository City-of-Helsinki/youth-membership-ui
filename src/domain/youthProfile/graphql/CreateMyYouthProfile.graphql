mutation CreateMyYouthProfile($input: CreateMyYouthProfileMutationInput!) {
  createMyYouthProfile(input: $input) {
    youthProfile {
      birthDate
    }
  }
}
