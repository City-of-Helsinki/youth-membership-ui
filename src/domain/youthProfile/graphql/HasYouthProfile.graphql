query HasYouthProfile {
  myYouthProfile {
    id
    membershipStatus
  }
}
