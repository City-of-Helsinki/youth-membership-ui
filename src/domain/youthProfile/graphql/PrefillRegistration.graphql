query PrefillRegistartion {
  myProfile {
    firstName
    lastName
    language
    primaryPhone {
      phone
      id
    }
    primaryAddress {
      address
      postalCode
      city
      countryCode
      id
      primary
      addressType
    }
    addresses {
      edges {
        node {
          primary
          id
          address
          postalCode
          city
          countryCode
          addressType
        }
      }
    }
    primaryEmail {
      email
    }
  }
}
