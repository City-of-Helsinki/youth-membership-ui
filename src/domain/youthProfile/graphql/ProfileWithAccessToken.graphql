query ProfileWithAccessToken($token: UUID!) {
  profileWithAccessToken(token: $token) {
    firstName
    lastName
    language
    id
    primaryAddress {
      address
      postalCode
      city
      countryCode
      id
      addressType
      primary
    }
    addresses {
      edges {
        node {
          primary
          id
          address
          postalCode
          city
          countryCode
          addressType
        }
      }
    }
    primaryEmail {
      email
      id
    }
    primaryPhone {
      phone
      id
    }
  }
}
