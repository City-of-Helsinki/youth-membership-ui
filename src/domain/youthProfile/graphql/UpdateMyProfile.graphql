mutation UpdateMyProfile($input: UpdateMyProfileMutationInput!) {
  updateMyProfile(input: $input) {
    profile {
      youthProfile {
        approverEmail
      }
    }
  }
}
