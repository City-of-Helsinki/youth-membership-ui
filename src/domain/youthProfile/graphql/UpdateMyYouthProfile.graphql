mutation UpdateMyYouthProfile($input: UpdateMyYouthProfileMutationInput!) {
    updateMyYouthProfile(input: $input) {
        youthProfile {
            approverEmail
        }
    }
}
