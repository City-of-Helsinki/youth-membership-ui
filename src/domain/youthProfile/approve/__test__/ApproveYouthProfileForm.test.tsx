import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { subYears, format } from 'date-fns';

import ApproveYouthProfileForm, {
  FormValues,
} from '../ApproveYouthProfileForm';
import { YouthLanguage } from '../../../../graphql/generatedTypes';

const defaultProps = {
  profile: {
    firstName: 'Teemu',
    lastName: 'Testaaja',
    birthDate: '2006-01-01',
    address: 'Testikuja 55, 00100 Helsinki',
    addresses: ['Kymintie 43, 00100 Helsinki'],
    email: 'teemu.testaaja@test.fi',
    languageAtHome: YouthLanguage.FINNISH,
    phone: '050123467',
    language: YouthLanguage.FINNISH,
    schoolClass: 'Luokka',
    schoolName: 'Koulu',
    photoUsageApproved: 'false',
    approverFirstName: 'Ville',
    approverLastName: 'Vanhempi',
    approverEmail: 'ville.vanhempi@test.fi',
    approverPhone: '05012345567',
  } as FormValues,
  isSubmitting: false,
  onSubmit: jest.fn(),
};

type Props = {
  profile?: FormValues;
};

const getWrapper = (props: Props) => {
  const combinedProps = {
    ...defaultProps,
    profile: {
      ...defaultProps.profile,
      ...props.profile,
    },
  };
  return mount(<ApproveYouthProfileForm {...combinedProps} />);
};

test('matches snapshot', () => {
  const wrapper = mount(<ApproveYouthProfileForm {...defaultProps} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('input fields are pre-filled', () => {
  const wrapper = mount(<ApproveYouthProfileForm {...defaultProps} />);
  const approverFirstName = wrapper.find('input[id="approverFirstName"]');
  const approverLastName = wrapper.find('input[id="approverLastName"]');
  const approverEmail = wrapper.find('input[id="approverEmail"]');
  const approverPhone = wrapper.find('input[id="approverPhone"]');

  expect(approverFirstName.props().value).toEqual('Ville');
  expect(approverLastName.props().value).toEqual('Vanhempi');
  expect(approverEmail.props().value).toEqual('ville.vanhempi@test.fi');
  expect(approverPhone.props().value).toEqual('05012345567');
});

describe('photoUsageTests', () => {
  test('child is under 15 and photoUsageApproved is shown', () => {
    const childAge = format(subYears(new Date(), 14), 'yyyy-MMM-dd');
    const wrapper = getWrapper({
      profile: { birthDate: childAge } as FormValues,
    });
    const photoUsage = wrapper.find('input[name="photoUsageApproved"]');
    expect(photoUsage).toBeTruthy();
  });

  test('child is 15 or older and photoUsageApproved is hidden', () => {
    const childAge = format(subYears(new Date(), 15), 'yyyy-MMM-dd');
    const wrapper = getWrapper({
      profile: { birthDate: childAge } as FormValues,
    });
    const photoUsage = wrapper.find('input[name="photoUsageApproved"]');
    expect(photoUsage.length).toEqual(0);
  });
});
