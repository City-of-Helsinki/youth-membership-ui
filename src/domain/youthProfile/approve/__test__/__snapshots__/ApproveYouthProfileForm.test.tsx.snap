// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="stack xl"
  >
    <div>
      <h1
        class="text h1 "
      >
        Hyväksy jäsenyys
      </h1>
      <p
        class="text info "
      >
        Teemu
         
        on lähettänyt hyväksyttäväksesi Helsingin kaupungin nuorisopalvelujen jäsenyyshakemuksen. Jäsenyys on maksuton. Hyväksymällä hakemuksen,
         
        Teemu
         
        voi viettää aikaa nuorisotaloilla, osallistua harrastusryhmiin, kursseille ja leireille Nuorisotaloilla toimintaa ohjaavat koulutetut nuoriso-ohjaajat. Mikäli tiedoissa on virheitä, pyydä nuorta muuttamaan tiedot ja lähettämään sinulle uusi varmistusviesti.
      </p>
    </div>
    <div>
      <h2
        class="text h2 "
      >
        Perustiedot
      </h2>
      <div
        class="grid"
      >
        <div
          class="wrapper noMargin"
        >
          <strong
            class="label"
          >
            Nimi
          </strong>
          <span
            class="value"
          >
            Teemu Testaaja
          </span>
        </div>
        <div
          class="wrapper noMargin"
        >
          <strong
            class="label"
          >
            Syntymäpäivä
          </strong>
          <span
            class="value"
          >
            01.01.2007
          </span>
        </div>
        <div
          class="wrapper noMargin"
        >
          <strong
            class="label"
          >
            Sähköposti
          </strong>
          <span
            class="value"
          >
            teemu.testaaja@test.fi
          </span>
        </div>
        <div
          class="wrapper noMargin"
        >
          <strong
            class="label"
          >
            Puhelinnumero
          </strong>
          <span
            class="value"
          >
            050123467
          </span>
        </div>
        <div
          class="wrapper noMargin"
        >
          <strong
            class="label"
          >
            Profiilin kieli
          </strong>
          <span
            class="value"
          >
            Suomi
          </span>
        </div>
        <div
          class="gridRow"
        >
          <div
            class="wrapper noMargin"
          >
            <strong
              class="label"
            >
              Osoite
            </strong>
            <span
              class="value"
            >
              Testikuja 55, 00100 Helsinki
            </span>
          </div>
          <div
            class="wrapper noMargin"
          >
            <strong
              class="label"
            >
              Osoite 2
            </strong>
            <span
              class="value"
            >
              Kymintie 43, 00100 Helsinki
            </span>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h2
        class="text h2 "
      >
        Lisätiedot
      </h2>
      <div
        class="grid"
      >
        <div
          class="wrapper noMargin"
        >
          <strong
            class="label"
          >
            Koulu ja luokka
          </strong>
          <span
            class="value"
          >
            Koulu, Luokka
          </span>
        </div>
        <div
          class="wrapper noMargin"
        >
          <strong
            class="label"
          >
            Kotona puhutut kielet
          </strong>
          <span
            class="value"
          >
            Suomi
          </span>
        </div>
      </div>
    </div>
    <form
      action="#"
    >
      <div
        class="stack xl"
      >
        <div>
          <h2
            class="text h2 "
          >
            Huoltajalta vaadittavat luvat
          </h2>
          <h3
            class="text h3 "
          >
            Kuvauslupa
          </h3>
          <p
            class="text info "
          >
            Lapsesta voi ottaa kuvia ja niitä saa käyttää Helsingin kaupungin viestinnässä ja markkinoinnissa painetussa ja sähköisessä mediassa. Kuvaamisesta ja kuvien käytöstä ei makseta korvausta. Oikeus kuvien käyttöön on aina voimassa ja koskee koko Helsingin kaupungin viestintää ja markkinointiviestintää.
          </p>
          <div
            class="formFields"
          >
            <ul
              class="list"
            >
              <li
                class="radioButtonRow"
              >
                <div
                  class="RadioButton-module_radioButton__1xM1I radio-button_hds-radio-button__1kktv"
                >
                  <input
                    class="RadioButton-module_input__1cTWc radio-button_hds-radio-button__input__27sH6"
                    id="photoUsageApprovedYes"
                    name="photoUsageApproved"
                    type="radio"
                    value="true"
                  />
                  <label
                    class="RadioButton-module_label__3Zij9 radio-button_hds-radio-button__label__1XnIa"
                    for="photoUsageApprovedYes"
                  >
                    Kyllä
                  </label>
                </div>
              </li>
              <li
                class="radioButtonRow"
              >
                <div
                  class="RadioButton-module_radioButton__1xM1I radio-button_hds-radio-button__1kktv"
                >
                  <input
                    checked=""
                    class="RadioButton-module_input__1cTWc radio-button_hds-radio-button__input__27sH6"
                    id="photoUsageApprovedNo"
                    name="photoUsageApproved"
                    type="radio"
                    value="false"
                  />
                  <label
                    class="RadioButton-module_label__3Zij9 radio-button_hds-radio-button__label__1XnIa"
                    for="photoUsageApprovedNo"
                  >
                    Ei
                  </label>
                </div>
              </li>
            </ul>
          </div>
        </div>
        <div>
          <h2
            class="text h2 "
          >
            Huoltajan tiedot
          </h2>
          <p
            class="text info "
          >
            Huoltajaan voidaan olla yhteydessä lasta koskevissa asioissa.
          </p>
          <div
            class="stack xl"
          >
            <div
              class="formGrid vertical-default"
            >
              <div
                class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq input"
              >
                <label
                  class="FieldLabel-module_label__1zrXK "
                  for="approverFirstName"
                >
                  Etunimi *
                </label>
                <div
                  class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
                >
                  <input
                    autocomplete="given-name"
                    class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
                    id="approverFirstName"
                    name="approverFirstName"
                    type="text"
                    value="Ville"
                  />
                </div>
              </div>
              <div
                class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq input"
              >
                <label
                  class="FieldLabel-module_label__1zrXK "
                  for="approverLastName"
                >
                  Sukunimi *
                </label>
                <div
                  class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
                >
                  <input
                    autocomplete="family-name"
                    class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
                    id="approverLastName"
                    name="approverLastName"
                    type="text"
                    value="Vanhempi"
                  />
                </div>
              </div>
              <div
                class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq input"
              >
                <label
                  class="FieldLabel-module_label__1zrXK "
                  for="approverEmail"
                >
                  Sähköpostiosoite *
                </label>
                <div
                  class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
                >
                  <input
                    autocomplete="email"
                    class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
                    id="approverEmail"
                    name="approverEmail"
                    type="email"
                    value="ville.vanhempi@test.fi"
                  />
                </div>
              </div>
              <div
                class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq input"
              >
                <label
                  class="FieldLabel-module_label__1zrXK "
                  for="approverPhone"
                >
                  Puhelinnumero *
                </label>
                <div
                  class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
                >
                  <input
                    autocomplete="tel"
                    class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
                    id="approverPhone"
                    name="approverPhone"
                    type="tel"
                    value="05012345567"
                  />
                </div>
              </div>
            </div>
            <div
              class="stack m"
            >
              <button
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_supplementary__3YKiS button_hds-button--supplementary__GcHcV addItemButton"
                type="button"
              >
                <div
                  aria-hidden="true"
                  class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <path
                        d="M0 0h24v24H0z"
                      />
                      <path
                        d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 3v4h4v2h-4v4h-2v-4H7v-2h4V7h2z"
                        fill="currentColor"
                      />
                    </g>
                  </svg>
                </div>
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  Lisää toisen huoltajan yhteystiedot
                </span>
              </button>
            </div>
          </div>
        </div>
        <div>
          <div
            class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
          >
            <input
              class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
              id="terms"
              name="terms"
              type="checkbox"
              value="false"
            />
            <label
              class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
              for="terms"
            >
              <span
                class="listLabel"
              >
                Olen tutustunut 
                <a
                  class="externalLink "
                  href="https://www.hel.fi/static/liitteet/kanslia/rekisteriselosteet/Kuva/Kuva-EU-Nuorisopalvelujen-jasenrekisteri.pdf"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  tämän palvelun rekisteriselosteeseen
                   
                  <svg
                    aria-label="Aukeaa uuteen ikkunaan"
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <path
                        d="M0 0h24v24H0z"
                      />
                      <path
                        d="M10 3v2H5v14h14v-5h2v7H3V3h7zm11 0v8h-2V6.413l-7 7.001L10.586 12l6.999-7H13V3h8z"
                        fill="currentColor"
                      />
                    </g>
                  </svg>
                </a>
                , 
                <a
                  class="externalLink "
                  href="https://www.hel.fi/static/liitteet/kanslia/rekisteriselosteet/Keha/Kanslia-EU-Sahkoisten-asiointipalveluiden-rekisteri.pdf"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Helsinki-profiilin rekisteriselosteeseen
                   
                  <svg
                    aria-label="Aukeaa uuteen ikkunaan"
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <path
                        d="M0 0h24v24H0z"
                      />
                      <path
                        d="M10 3v2H5v14h14v-5h2v7H3V3h7zm11 0v8h-2V6.413l-7 7.001L10.586 12l6.999-7H13V3h8z"
                        fill="currentColor"
                      />
                    </g>
                  </svg>
                </a>
                 ja 
                <a
                  class="externalLink "
                  href="https://www.hel.fi/helsinki/fi/kaupunki-ja-hallinto/tietoa-helsingista/tietosuoja/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  kaupungin tietosuojakäytäntöön
                   
                  <svg
                    aria-label="Aukeaa uuteen ikkunaan"
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <path
                        d="M0 0h24v24H0z"
                      />
                      <path
                        d="M10 3v2H5v14h14v-5h2v7H3V3h7zm11 0v8h-2V6.413l-7 7.001L10.586 12l6.999-7H13V3h8z"
                        fill="currentColor"
                      />
                    </g>
                  </svg>
                </a>
                . Minuun voidaan olla yhteydessä antamieni tietojen avulla.
              </span>
            </label>
          </div>
        </div>
        <button
          class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO button"
          type="submit"
        >
          <span
            class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
          >
            Hyväksy jäsenyys
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
